### PR TITLE
Pull request to merge from dev-gmx to dev-frontend

### PR DESCRIPTION
**Pull request by Carl26**

### CODE REVIEW

**Comment by Carl26** [comment]
This update only contains the new forgot password feature and some minor improvements

File: frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/ResetPasswordActivity.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        unregisterReceiver(resetPwdReceiver);
+    }
+}
+
+class ResetPwdWorkingThread extends AsyncTask<String, Void, Void> {
+    String vCodeUrl = "http://128.199.231.190:8080/api/users/resetPassword";
```
Comment (Sheng-Xuan) [inline frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/ResetPasswordActivity.java:132]: Actually we should make it consistent to use Singleton instance of httpbuilder, easier to maintain.

File: frontend/Movie_Pigeon/app/src/main/res/layout/activity_reset_password.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        android:textSize="25dp"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="50dp"
+        android:id="@+id/tvResetPassword" />
+
+    <EditText
+        android:layout_width="200dp"
+        android:id="@+id/rsETEmail"
+        android:cursorVisible="true"
+        android:textSize="13dp"
```
Comment (Sheng-Xuan) [inline frontend/Movie_Pigeon/app/src/main/res/layout/activity_reset_password.xml:18]: As mentioned in my pr, textSize should be in sp.

File: frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/GlobalReceiver.java [hunk 3]
```diff
@@ -60,6 +60,11 @@ public void onReceive(Context context, Intent intent) {
                 sBuilder.execute(signInDetails);
                 break;
 
+            case "killRegistration":
+                Log.i(TAG, "Kill Registration received");
+                ((Activity) context).finish();
+                break;
+
             case "UserUpdate":
```
Comment (Sheng-Xuan) [inline frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/GlobalReceiver.java:68]: to be consistent, this should be 'userUpdate'. Or change others to start with a capital letter.

File: frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegistrationActivity.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        unregisterReceiver(globalReceiver);
+    }
+
+    private int emailChecker(String email) {
```
Comment (Sheng-Xuan) [inline frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/RegistrationActivity.java:116]: checker can improve:
1.Only one '@' is in the string.
2.pattern of *.* after @.

**Review by Sheng-Xuan — APPROVED** [review]

**Pull request merged**